### ISSUE TITLE
Add the TensorBoard-tracked MVE regression example

### ISSUE THREAD

**Issue by debrevitatevitae** [GroupiSP/probreg#31]
## Parent

Spec: #27

## What to build

A user new to probreg runs one command, trains a deliberately trivial one-dimensional mean-variance model, and watches the whole run in TensorBoard:

```
python examples/jax/tracking/run.py --logdir runs/
```

Opening TensorBoard on that directory, they find per-epoch training and validation scalars under stage-namespaced tags, the run's hyperparameters together with the identities of the loss and the metrics under the HParams plugin, and a final figure showing the fitted mean with its predictive interval. Each invocation writes to its own timestamped subdirectory, so successive runs appear side by side rather than overwriting one another.

The regression problem is the existing MVE example's configuration reused verbatim — heteroscedastic noise on a linear mean, held-out validation, early stopping on validation loss, best-model checkpointing — with early-stopping patience raised so the tracked curves are long enough to be worth looking at. The example keeps its printing sink alongside the tracker sink: tracking is additive, not a replacement for what the user already sees in the terminal.

The tracker itself is a small module the user is meant to copy. It is backend-neutral: it accepts floats, arrays and figures, never probreg types, so nothing about it is specific to the JAX backend. Its writer is injected through the constructor and tensorboardX is imported lazily, which is what makes its mapping logic testable without the dependency installed. `log_artifact` dispatches on the value's type — figure to image, string to text, anything else a `TypeError` rather than a silent drop.

TensorBoard and its writer arrive as a dependency group, never as a package extra: the library gains no new dependency, hard or optional, from this ticket.

## Acceptance criteria

- [ ] A tracking example directory holds the tracker module, the runner, an `__init__.py` and a README; no module in it is named `tensorboard.py`
- [ ] The tracker satisfies `ExperimentTracker`, takes its writer by constructor injection, imports tensorboardX lazily, and accepts only backend-neutral values
- [ ] `log_params` records hyperparameters plus the loss and metric identities via the HParams plugin, flattening nested values
- [ ] `log_artifact` dispatches by type and raises `TypeError` on an unsupported value
- [ ] The runner exposes `main(argv: Sequence[str] | None = None)` and accepts `--logdir`, defaulting to `runs/`, writing each run under a UTC-timestamped subdirectory
- [ ] Both the printing sink and `TrackerEventSink` are passed to the run
- [ ] Held-out validation is configured with an empty metric prefix so the tracker owns the whole tag
- [ ] The final figure draws the mean with a `1.96 * scale` predictive interval, uses a non-interactive backend, and never opens a plot window
- [ ] TensorBoard and tensorboardX are declared as a dependency group, not an extra
- [ ] The README states how to install the group, run the example, and launch TensorBoard
- [ ] Tested at the tracker's constructor seam with a fake writer, requiring neither tensorboardX nor JAX: type dispatch in `log_artifact` and injectivity of parameter flattening
- [ ] Tested end to end by calling the runner's `main(argv)` under `importorskip`, following the existing example-test pattern
- [ ] `pre-commit run --all-files` and `pytest` both pass

## Blocked by

- #28
- #29
